core-plugins/src/main/java/co/cask/hydrator/plugin/error/ErrorCollector.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.error;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.EndpointPluginContext;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.ErrorRecord;
+import co.cask.cdap.etl.api.ErrorTransform;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.ws.rs.Path;
+
+/**
+ * Adds the error code and error message to each record, then emits it.
+ */
+@Plugin(type = ErrorTransform.PLUGIN_TYPE)
+@Name("ErrorCollector")
+public class ErrorCollector extends ErrorTransform<StructuredRecord, StructuredRecord> {
+  private final Config config;
+
+  public ErrorCollector(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      if (inputSchema.getField(config.messageField) != null) {
```

Comment:
can this produce NPE is config.messageField is not set? and the same for the next two